app/src/main/java/io/appium/uiautomator2/server/mjpeg/MjpegScreenshotStream.java

```diff
@@ -53,8 +56,8 @@ public void interrupt() {
     @Override
     public void run() {
         while (!isStopped) {
-            if (clients.size() == 0) {
-               SystemClock.sleep(500);
+            if (clients.isEmpty()) {
```

Comment:
please add synchronisation. Something like:

```
boolean hasNoClients;
synchronized (clients) {
  hasNoClients = clients.isEmpty();
}
if (hasNoClients) ...
```

Reply:
This will be resolved w/ the upcoming update(s)